pkg/server/api/middleware/caller.go

```diff
@@ -0,0 +1,68 @@
+package middleware
+
+import (
+	"context"
+
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/pkg/server/api/rpccontext"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+)
+
+func callerContextFromContext(ctx context.Context) (context.Context, error) {
+	p, ok := peer.FromContext(ctx)
+	if !ok {
+		return nil, status.Error(codes.Internal, "no peer information available")
```

Comment:
Looks like we currently return `codes.Unauthenticated` in this case. Curious to know if you consider that needs to be changed?

Reply:
This check is purely defensive. I don't expect this to ever happen, since gRPC takes care of adding the peer to the context. That being said,  I think `Internal` is a clearer code and communicates that something is really wrong internally. A caller might interpret `Unauthenticated` to mean that they could take some action to become authenticated, but in this case, there isn't anything they can do.